internal/interpreter/api.rs

```diff
@@ -967,24 +971,51 @@ impl ComponentInstance {
             .map_err(|()| SetCallbackError::NoSuchCallback)
     }
 
-    /// Call the given callback within a global singleton with the arguments
+    /// Call the given callback or function within a global singleton with the arguments
     ///
-    /// ## Examples
-    /// See the documentation of [`Self::set_global_callback`] for an example
+    /// Renamed to `invoke_global()`
+    #[deprecated(note = "renamed to invoke_global")]
     pub fn invoke_global_callback(
         &self,
         global: &str,
         callback_name: &str,
         args: &[Value],
-    ) -> Result<Value, InvokeCallbackError> {
+    ) -> Result<Value, InvokeError> {
+        self.invoke_global(global, callback_name, args)
+    }
+    /// Call the given callback or function within a global singleton with the arguments
+    ///
+    /// ## Examples
+    /// See the documentation of [`Self::set_global_callback`] for an example
+    pub fn invoke_global(
+        &self,
+        global: &str,
+        callable_name: &str,
+        args: &[Value],
+    ) -> Result<Value, InvokeError> {
         generativity::make_guard!(guard);
         let comp = self.inner.unerase(guard);
-        comp.description()
+        let g = comp
+            .description()
             .get_global(comp.borrow(), &normalize_identifier(global))
-            .map_err(|()| InvokeCallbackError::NoSuchCallback)? // FIXME: should there be a NoSuchGlobal error?
-            .as_ref()
-            .invoke_callback(&normalize_identifier(callback_name), args)
-            .map_err(|()| InvokeCallbackError::NoSuchCallback)
+            .map_err(|()| InvokeError::NoSuchCallable)?; // FIXME: should there be a NoSuchGlobal error?
+        if matches!(
+            comp.description()
+                .original
+                .root_element
+                .borrow()
+                .lookup_property(callable_name)
+                .property_type,
+            LangType::Function { .. }
+        ) {
+            g.as_ref()
+                .eval_function(callable_name, args.iter().cloned().collect())
```

Comment:
Oh there's no need to normalize for functions?

Reply:
well spotted!